opentelemetry-proto/src/transform/logs.rs

```diff
@@ -0,0 +1,332 @@
+                            value: Some(AnyValue {
+                                value: Some(value.into()),
+                            }),
+                        })
+                        .collect(),
+                }),
+                Any::Bytes(v) => Value::BytesValue(v),
+            }
+        }
+    }
+
+    impl From<Severity> for SeverityNumber {
+        fn from(number: Severity) -> Self {
+            match number {
+                Severity::Trace => SeverityNumber::Trace,
+                Severity::Trace2 => SeverityNumber::Trace2,
+                Severity::Trace3 => SeverityNumber::Trace3,
+                Severity::Trace4 => SeverityNumber::Trace4,
+                Severity::Debug => SeverityNumber::Debug,
+                Severity::Debug2 => SeverityNumber::Debug2,
+                Severity::Debug3 => SeverityNumber::Debug3,
+                Severity::Debug4 => SeverityNumber::Debug4,
+                Severity::Info => SeverityNumber::Info,
+                Severity::Info2 => SeverityNumber::Info2,
+                Severity::Info3 => SeverityNumber::Info3,
+                Severity::Info4 => SeverityNumber::Info4,
+                Severity::Warn => SeverityNumber::Warn,
+                Severity::Warn2 => SeverityNumber::Warn2,
+                Severity::Warn3 => SeverityNumber::Warn3,
+                Severity::Warn4 => SeverityNumber::Warn4,
+                Severity::Error => SeverityNumber::Error,
+                Severity::Error2 => SeverityNumber::Error2,
+                Severity::Error3 => SeverityNumber::Error3,
+                Severity::Error4 => SeverityNumber::Error4,
+                Severity::Fatal => SeverityNumber::Fatal,
+                Severity::Fatal2 => SeverityNumber::Fatal2,
+                Severity::Fatal3 => SeverityNumber::Fatal3,
+                Severity::Fatal4 => SeverityNumber::Fatal4,
+            }
+        }
+    }
+
+    fn attributes_to_keyvalue(attributes: BTreeMap<Cow<'static, str>, Any>) -> Vec<KeyValue> {
+        attributes
+            .into_iter()
+            .map(|(key, value)| KeyValue {
+                key: key.into_owned(),
+                value: Some(AnyValue {
+                    value: Some(value.into()),
+                }),
+            })
+            .collect()
+    }
+
+    impl From<opentelemetry::sdk::log::LogRecord> for LogRecord {
+        fn from(log_record: opentelemetry::sdk::log::LogRecord) -> Self {
+            let trace_context = log_record.trace_context.as_ref();
+
+            let record = LogRecord {
+                time_unix_nano: log_record.timestamp.map(to_nanos).unwrap_or_default(),
+                observed_time_unix_nano: log_record
+                    .observed_timestamp
+                    .map(to_nanos)
+                    .unwrap_or_default(),
+                severity_number: log_record
+                    .severity_number
+                    .map(SeverityNumber::from)
+                    .map(Into::into)
```

Comment:
Can we implement a conversion between the `serverity_number` type in `log_record` and the grpc generated types here? Two mappings here seem to be unnecessary.

Reply:
Yup, moved the mapping code to the function(s) itself.